### PR TITLE
Add proximity matrix feature extraction

### PR DESCRIPTION
**Pull request by fineguy**

### CODE REVIEW

File: ast2vec/repo2/proxbase.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        else:
+            n_roles = 0
+
+        if tokens_to_tokens or roles_to_tokens:
+            n_tokens = len(token2ind)
+        else:
+            n_tokens = 0
+
+        n_nodes = n_roles + n_tokens
+        # print(n_roles, n_tokens)
```
Comment (EgorBu) [inline ast2vec/repo2/proxbase.py]: please remove comments

File: ast2vec/repo2/proxbase.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        fill_mat.count = 0
+
+        fill_mat(roles_to_roles, (0, 0))
+        fill_mat(roles_to_tokens, (0, n_roles))
+        fill_mat(tokens_to_tokens, (n_roles, n_roles))
+
+        mat = coo_matrix(mat + mat.T - diags(mat.diagonal()))
+        return mat
+
+    def _traverse_uast(self, root, roles, tokens, role2ind, token2ind, dok_mat):
```
Comment (EgorBu) [inline ast2vec/repo2/proxbase.py]: Please add docstring

File: ast2vec/__init__.py [hunk 3]
```diff
@@ -7,8 +7,10 @@
 from ast2vec.repo2.base import Repo2Base, ensure_bblfsh_is_running_noexc
 from ast2vec.repo2.coocc import Repo2Coocc, Repo2CooccTransformer
 from ast2vec.repo2.nbow import Repo2nBOW, Repo2nBOWTransformer
+from ast2vec.repo2.prox import Repo2ProxSwivel
 from ast2vec.repo2.source import Repo2Source, Repo2SourceTransformer
 from ast2vec.source import Source
 
+
```
Comment (vmarkovtsev) [inline ast2vec/__init__.py]: Remove this new line - not needed

File: ast2vec/__main__.py [hunk 4]
```diff
@@ -13,6 +13,8 @@
 from ast2vec.repo2.base import Repo2Base
 from ast2vec.repo2.coocc import repo2coocc_entry, repos2coocc_entry
 from ast2vec.repo2.nbow import repo2nbow_entry, repos2nbow_entry
+from ast2vec.repo2.prox import repo2prox_entry, _MAT_TYPES_
+from ast2vec.repo2.proxbase import _EDGE_TYPES_
```
Comment (vmarkovtsev) [inline ast2vec/__main__.py]: Wow. Let it be `EDGE_TYPES`

File: ast2vec/__main__.py [hunk 5]
```diff
@@ -13,6 +13,8 @@
 from ast2vec.repo2.base import Repo2Base
 from ast2vec.repo2.coocc import repo2coocc_entry, repos2coocc_entry
 from ast2vec.repo2.nbow import repo2nbow_entry, repos2nbow_entry
+from ast2vec.repo2.prox import repo2prox_entry, _MAT_TYPES_
```
Comment (vmarkovtsev) [inline ast2vec/__main__.py]: `MATRIX_TYPES`

File: ast2vec/__main__.py [hunk 6]
```diff
@@ -144,6 +146,38 @@ def main():
...
+        "repo2prox", formatter_class=argparse.RawTextHelpFormatter,
+        help="Produce the proximity matrix from a Git repository.")
+    repo2prox_parser.set_defaults(handler=repo2prox_entry)
+    repo2prox_parser.add_argument(
+        "-r", "--repository", required=True,
+        help="URL or path to a Git repository.")
+    repo2prox_parser.add_argument(
+        "--linguist", help="Path to src-d/enry executable.")
+    repo2prox_parser.add_argument(
+        "-m", "--mat-type", required=True, choices=_MAT_TYPES_.keys(),
```
Comment (vmarkovtsev) [inline ast2vec/__main__.py]: Better `--matrix-type`

**Pull request closed without merging**